spec/unit/provider/package/yum_spec.rb

```diff
@@ -248,6 +248,93 @@
       @provider = Chef::Provider::Package::Yum.new(@new_resource, @run_context)
       @provider.load_current_resource
       expect(@new_resource.package_name).to eq("test-package")
+      expect(@new_resource.version).to eq(nil)
+    end
+
+    it "should search provides if a package name can't be found then set version to match if a requirement was passed in the package name" do
```

Comment:
That makes sense I think - got a bunch of meetings this afternoon but will tackle that after or tomorrow morning if not.